media/js/careers/testimonials-modal.es6.js

```diff
@@ -0,0 +1,226 @@
+
+content.classList.add('c-careers-full-testimonial-content');
+
+// Setting up directional buttons
+const modalNextButtonFragment = `<div class="c-modal-next">
+        <button type="button" class="c-modal-button-next" title="Next">
+            Next
+        </button>
+    </div>`;
+
+const modalPrevButtonFragment = `<div class="c-modal-prev">
+        <button type="button" class="c-modal-button-prev" title="Previous">
+            Previous
+        </button>
+    </div>`;
+
+// Set to true when a listener it set, as to not set it multiple times.
+// Resets when the modal is closed.
+let kbInit = null;
+
+function keyboardListenerInit() {
+    if (kbInit) {
+        return;
+        // Turn Off
+    }
+
+    document.addEventListener('keyup', keyboardNextPrev, false);
+    kbInit = true;
+}
+
+// Keyboard arrows
+function keyboardNextPrev(event) {
+    if (!kbInit) {
+        return;
+    }
+
+    switch (event.keyCode) {
+        case 37: // Left arrow
+            prevModalArticle();
+            break;
+        case 38: // Up arrow
+            prevModalArticle();
+            break;
+        case 39: // Right arrow
+            nextModalArticle();
+            break;
+        case 40: // Down arrow
+            nextModalArticle();
+            break;
+    }
+}
+
+// Modal initializer
+function modalInit() {
+    const modalNextButton = document.querySelector('.c-modal-next');
+    const modalPrevButton = document.querySelector('.c-modal-prev');
+
+    modalNextButton.removeEventListener('click', nextModalArticle, false);
+    modalPrevButton.removeEventListener('click', prevModalArticle, false);
+
+    modalNextButton.addEventListener('click', nextModalArticle, false);
+    modalPrevButton.addEventListener('click', prevModalArticle, false);
+
+    keyboardListenerInit();
+}
+
+// Retrieve modal index number
+function getCurrentModalIndex() {
+    const modalContent = document.querySelector(
+        '.mzp-u-modal-content.mzp-c-modal-overlay-contents'
+    );
+    const newArticleIndex = parseInt(
+        modalContent.getAttribute('data-current-index'),
+        10
+    );
+    return newArticleIndex;
+}
+
+function updateModalArticle(index) {
+    const modalContent = document.querySelector(
+        '.mzp-u-modal-content.mzp-c-modal-overlay-contents'
+    );
+    const newArticleId = articleArray[index].getAttribute('data-modal-id');
+    const newModalContent = document
+        .querySelector(`[data-modal-parent="${newArticleId}"]`)
+        .cloneNode(true);
+    const currentModalContent = modalContent.firstElementChild;
+
+    window.location.hash = newArticleId;
+
+    modalContent.replaceChild(newModalContent, currentModalContent);
+    modalContent.setAttribute('data-current-index', index);
+
+    modalInit();
+}
+
+function nextModalArticle() {
+    let newArticleIndex = getCurrentModalIndex();
+    newArticleIndex++;
+
+    // If at the end of the gallery, start over
+    if (newArticleIndex === articleArray.length) {
```

Comment:
**Question** I _think_ that this should be `articleArray.length -1`? 

Ex: 
```js
let arr = [1, 2, 4, 5, 6];
console.log(arr.length);
-> 5
console.log(arr[arr.length]);
-> undefined
console.log(arr[arr.length -1]);
-> 6
```

Reply:
I don't think so. If we place `-1` at the end of that if-statement, then it skips over the last item in that array, because instead of counting up to `5` it is just counting up to `4`!